File: src/subside/plugins/koth/commands/CommandNext.java
package subside.plugins.koth.commands;

import org.bukkit.command.CommandSender;

import subside.plugins.koth.commands.CommandHandler.CommandCategory;
import subside.plugins.koth.exceptions.CommandMessageException;
import subside.plugins.koth.modules.Lang;
import subside.plugins.koth.scheduler.Schedule;
import subside.plugins.koth.utils.MessageBuilder;
import subside.plugins.koth.utils.Perm;

public class CommandNext extends AbstractCommand {

    public CommandNext(CommandCategory category) {
        super(category);
    }

    @Override
    public void run(CommandSender sender, String[] args) {
        Schedule schedule = getPlugin().getScheduleHandler().getNextEvent();
        if(schedule != null)
            throw new CommandMessageException(new MessageBuilder(Lang.COMMAND_NEXT_MESSAGE).koth(getPlugin().getKothHandler(), schedule.getKoth()).timeTillNext(schedule));

        throw new CommandMessageException(new MessageBuilder(Lang.COMMAND_NEXT_NO_NEXT_FOUND));
    }

    @Override
    public Perm getPermission() {
        return Perm.NEXT;
    }

    @Override
    public String[] getCommands() {
        return new String[]{"next"};
    }
    
    @Override
    public String getUsage() {
        return "/koth next";
    }

    @Override
    public String getDescription() {
        return "info about the next upcoming KoTH";
    }

}

File: src/subside/plugins/koth/commands/basic/CommandNext.kt
package subside.plugins.koth.commands.basic

import org.bukkit.command.CommandSender
import subside.plugins.koth.KothManager
import subside.plugins.koth.commands.Command
import subside.plugins.koth.modules.Lang
import subside.plugins.koth.utils.IPerm
import subside.plugins.koth.utils.MessageBuilder
import subside.plugins.koth.utils.Perm

class CommandNext : Command {
    override fun run(kothManager: KothManager, sender: CommandSender, args: Array<String>) {
        val schedule = kothManager.scheduleHandler.nextEvent

        if (schedule == null) {
            MessageBuilder(Lang.COMMAND_NEXT_NO_NEXT_FOUND).send(sender)
            return
        }

        MessageBuilder(Lang.COMMAND_NEXT_MESSAGE)
                .koth(kothManager.kothHandler, schedule.koth)
                .timeTillNext(schedule)
                .send(sender)
    }

    override val permission = Perm.NEXT
    override val commands = arrayOf("next")
    override val usage = "/koth next"
    override val description = "Info about the next upcoming KoTH"

}